spring-amqp/src/main/java/org/springframework/amqp/core/MessageListener.java

```diff
@@ -49,7 +49,9 @@ default void containerAckMode(AcknowledgeMode mode) {
 	 * @since 2.2
 	 */
 	default void onMessageBatch(List<Message> messages) {
-		throw new UnsupportedOperationException("Should never be called by the container");
+		throw new UnsupportedOperationException(
+				"The container has been configured with 'consumerBatchEnabled' but the listener is not "
```

Comment:
I don't think the `consumerBatchEnabled` context makes sense here.
It is really not a listener contract resposibility to worry about a container state...

Reply:
Well, "should never be called by the container" doesn't really help a user fix the problem.

What do you suggest?

Reply:
Have an assertion in the container. And here in the `UnsupportedOperationException` just mentioned what to use instead.

Reply:
I don't want to add an assertion in the happy path. Reverted to a generic message.

Reply:
Not needed - assertion already done in `doInitialize()`.